### PR TITLE
Do not hard-fail if formatting failed #23

### LINKED ISSUE #23: GitLab CI windows / webview2-com-sys: failed to run custom build command
Running the build for webview2-com-sys fails on GitLab CI fails with the error `Io(Error { kind: NotFound, message: "program not found" } )`. Unfortunately it doesn't say which program was not found. There also seems to not be a list of build dependencies in the README.

The minimal reproducible .gitlab-ci.yml is this: 

```yaml
stages:
  - build

build-job:
  stage: build
  image: rust
  tags:
  - shared-windows
  - windows
  - windows-1809
  script:
    - choco install -y rust
    - cargo new foo
    - cd foo
    - cargo add webview2-com-sys
    - cargo build
```

This is the full output:

```
   Compiling webview2-com-sys v0.26.0
error: failed to run custom build command for `webview2-com-sys v0.26.0`
Caused by:
  process didn't exit successfully: `C:\GitLab-Runner\builds\danyel\sky-fi\blabla\target\debug\build\webview2-com-sys-d3e6eca53f705af0\build-script-build` (exit code: 1)
  --- stdout
  cargo:rustc-link-search=native=C:\GitLab-Runner\builds\danyel\sky-fi\blabla\target\debug\build\webview2-com-sys-99dd0865e5d8b1a6\out\Microsoft.Web.WebView2.1.0.1938.49\build\native\x64
    Finished writing `C:\GitLab-Runner\builds\danyel\sky-fi\blabla\target\debug\build\webview2-com-sys-99dd0865e5d8b1a6\out\Microsoft.rs` in 1.95s
  --- stderr
  Copy from "C:\\GitLab-Runner\\builds\\danyel\\sky-fi\\blabla\\target\\debug\\build\\webview2-com-sys-99dd0865e5d8b1a6\\out\\Microsoft.Web.WebView2.1.0.1938.49\\build\\native\\arm64\\WebView2Loader.dll" -> "C:\\Users\\gitlab_runner\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\webview2-com-sys-0.26.0\\arm64\\WebView2Loader.dll"
  Copy from "C:\\GitLab-Runner\\builds\\danyel\\sky-fi\\blabla\\target\\debug\\build\\webview2-com-sys-99dd0865e5d8b1a6\\out\\Microsoft.Web.WebView2.1.0.1938.49\\build\\native\\arm64\\WebView2Loader.dll.lib" -> "C:\\Users\\gitlab_runner\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\webview2-com-sys-0.26.0\\arm64\\WebView2Loader.dll.lib"
  Copy from "C:\\GitLab-Runner\\builds\\danyel\\sky-fi\\blabla\\target\\debug\\build\\webview2-com-sys-99dd0865e5d8b1a6\\out\\Microsoft.Web.WebView2.1.0.1938.49\\build\\native\\arm64\\WebView2LoaderStatic.lib" -> "C:\\Users\\gitlab_runner\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\webview2-com-sys-0.26.0\\arm64\\WebView2LoaderStatic.lib"
  Copy from "C:\\GitLab-Runner\\builds\\danyel\\sky-fi\\blabla\\target\\debug\\build\\webview2-com-sys-99dd0865e5d8b1a6\\out\\Microsoft.Web.WebView2.1.0.1938.49\\build\\native\\x64\\WebView2Loader.dll" -> "C:\\Users\\gitlab_runner\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\webview2-com-sys-0.26.0\\x64\\WebView2Loader.dll"
  Copy from "C:\\GitLab-Runner\\builds\\danyel\\sky-fi\\blabla\\target\\debug\\build\\webview2-com-sys-99dd0865e5d8b1a6\\out\\Microsoft.Web.WebView2.1.0.1938.49\\build\\native\\x64\\WebView2Loader.dll.lib" -> "C:\\Users\\gitlab_runner\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\webview2-com-sys-0.26.0\\x64\\WebView2Loader.dll.lib"
  Copy from "C:\\GitLab-Runner\\builds\\danyel\\sky-fi\\blabla\\target\\debug\\build\\webview2-com-sys-99dd0865e5d8b1a6\\out\\Microsoft.Web.WebView2.1.0.1938.49\\build\\native\\x64\\WebView2LoaderStatic.lib" -> "C:\\Users\\gitlab_runner\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\webview2-com-sys-0.26.0\\x64\\WebView2LoaderStatic.lib"
  Copy from "C:\\GitLab-Runner\\builds\\danyel\\sky-fi\\blabla\\target\\debug\\build\\webview2-com-sys-99dd0865e5d8b1a6\\out\\Microsoft.Web.WebView2.1.0.1938.49\\build\\native\\x86\\WebView2Loader.dll" -> "C:\\Users\\gitlab_runner\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\webview2-com-sys-0.26.0\\x86\\WebView2Loader.dll"
  Copy from "C:\\GitLab-Runner\\builds\\danyel\\sky-fi\\blabla\\target\\debug\\build\\webview2-com-sys-99dd0865e5d8b1a6\\out\\Microsoft.Web.WebView2.1.0.1938.49\\build\\native\\x86\\WebView2Loader.dll.lib" -> "C:\\Users\\gitlab_runner\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\webview2-com-sys-0.26.0\\x86\\WebView2Loader.dll.lib"
  Copy from "C:\\GitLab-Runner\\builds\\danyel\\sky-fi\\blabla\\target\\debug\\build\\webview2-com-sys-99dd0865e5d8b1a6\\out\\Microsoft.Web.WebView2.1.0.1938.49\\build\\native\\x86\\WebView2LoaderStatic.lib" -> "C:\\Users\\gitlab_runner\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\webview2-com-sys-0.26.0\\x86\\WebView2LoaderStatic.lib"
  Error: Io(Error { kind: NotFound, message: "program not found" })
Cleaning up project directory and file based variables 00:03
ERROR: Job failed: exit status 1
```

### PR DESCRIPTION
**Pull request by muja**
Closes #23

### CODE REVIEW

File: crates/bindings/build.rs [hunk 1]
```diff
@@ -341,7 +341,9 @@ mod webview2_bindgen {
 
     pub fn update_bindings() -> super::Result<bool> {
         let source_path = generate_bindings()?;
-        format_bindings(&source_path)?;
```
[the remark this answers is not in the document]
Comment (wravery) [reply crates/bindings/build.rs:344]: In other words, skip updating the bindings entirely if they cannot be formatted.

**Review by wravery — CHANGES_REQUESTED** [review]

File: crates/bindings/build.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (wravery) [reply crates/bindings/build.rs:344]: For example, in `main()`:
```rust
    if let Err(err) = webview2_bindgen::update_bindings() {
        eprintln!("Error updating bindings, using previous version: {err}");
    }
```

[the remark this answers is not in the document]
Comment (muja) [reply crates/bindings/build.rs:344]: I don't see the problem. Why would I (as a user) care about formatting in some build directory?
That is strictly a contributor's problem, they should be instructed to not create unformatted PRs

**Comment by wravery** [comment]
Thank you for the PR! I decided to go with a different approach, though, removing all of the build-time code gen and making that a separate, explicit `update-bindings` tool to run when updating the `winmd` file.

**Comment by muja** [comment]
Good job. I wasn't thinking you'd be so fast, of course that's more sensible. Thank you very much!

**Pull request closed without merging**